iothub/service/src/DigitalTwin/DigitalTwinsClient.cs

```diff
@@ -0,0 +1,363 @@
+
+        /// <summary>
+        /// Creates an instance of this class. Provided for unit testing purposes only.
+        /// </summary>
+        protected DigitalTwinsClient()
+        {
+        }
+
+        internal DigitalTwinsClient(string hostName, IotHubConnectionProperties credentialProvider, HttpClient httpClient, HttpRequestMessageFactory httpRequestMessageFactory)
+        {
+            _hostName = hostName;
+            _credentialProvider = credentialProvider;
+            _httpClient = httpClient;
+            _httpRequestMessageFactory = httpRequestMessageFactory;
+        }
+
+        /// <summary>
+        /// Gets a strongly-typed digital twin.
+        /// </summary>
+        /// <param name="digitalTwinId">The Id of the digital twin.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>The deserialized application/json digital twin and the ETag for the digital twin.</returns>
+        /// <exception cref="ArgumentNullException">Thrown when the provided digital twin id is null.</exception>
+        /// <exception cref="ArgumentException">Thrown when the provided digital twin id is empty or whitespace.</exception>
+        /// <exception cref="IotHubException">
+        /// Thrown if IoT hub responded to the request with a non-successful status code. For example, if the provided
+        /// request was throttled, <see cref="IotHubThrottledException"/> is thrown. For a complete list of possible
+        /// error cases, see <see cref="Common.Exceptions"/>.
+        /// </exception>
+        /// <exception cref="HttpRequestException">
+        /// If the HTTP request fails due to an underlying issue such as network connectivity, DNS failure, or server
+        /// certificate validation.
+        /// </exception>
+        /// <exception cref="OperationCanceledException">If the provided cancellation token has requested cancellation.</exception>
+        public virtual async Task<GetDigitalTwinResponse<T>> GetAsync<T>(string digitalTwinId, CancellationToken cancellationToken = default)
+        {
+            if (Logging.IsEnabled)
+                Logging.Enter(this, $"Getting digital twin with Id: {digitalTwinId}", nameof(GetAsync));
+
+            try
+            {
+                Argument.RequireNotNullOrEmpty(digitalTwinId, nameof(digitalTwinId));
+
+                cancellationToken.ThrowIfCancellationRequested();
+
+                using HttpRequestMessage request = _httpRequestMessageFactory.CreateRequest(HttpMethod.Get, GetDigitalTwinRequestUri(digitalTwinId), _credentialProvider);
+                HttpResponseMessage response = await _httpClient.SendAsync(request, cancellationToken).ConfigureAwait(false);
+                await HttpMessageHelper2.ValidateHttpResponseStatus(HttpStatusCode.OK, response).ConfigureAwait(false);
+                T digitalTwin = await HttpMessageHelper2.DeserializeResponse<T>(response, cancellationToken).ConfigureAwait(false);
```

Comment:
Same question for this deserializer as well - why is it async?

Reply:
This helper method reads the payload of the http response with ```response.Content.ReadHttpContentAsStringAsync(...)```

Would it be better to do this synchronously?

Reply:
Ah, I see. No, what you have here is good; we don't want to block on an operation when an async version exists. 
Another option is to directly read the response content once, and then pass in the string to the validate and deserialize methods. We should likely also see some improvement with speed, although I think it'll be quite small (since we're only making 2 calls per API).

Reply:
The ```ValidateHttpResponseStatus``` call doesn't read the body unless an error code is attached to the response, so we woudn't save any payload read operations by factoring it out here. We either read it once before throwing an exception or we read it once before returning the successful result